### PR TITLE
Clean PrecisionNum

### PR DESCRIPTION
**Pull request by nimo23**
Changes proposed in this pull request:
- use log.error instead of printstacktrace..
- use "BigDecimal.valueOf(double)" instead of "new BigDecimal(double)"
- minor cleanings..
- [ ] added an entry to the unreleased section of `CHANGES.md`

### CODE REVIEW

**Comment by nimo23** [comment]
why a mathContext for floating poins in BigDecimal?

```
private PrecisionNum(float val) {
        mathContext = new MathContext(DEFAULT_PRECISION, RoundingMode.HALF_UP);
        delegate = new BigDecimal(val, mathContext);
    }

    private PrecisionNum(double val) {
        mathContext = new MathContext(DEFAULT_PRECISION, RoundingMode.HALF_UP);
        delegate = BigDecimal.valueOf(val);
    }
```

Should I replace this by:

```
private PrecisionNum(float val) {
        delegate = BigDecimal.valueOf(val);
    }
private PrecisionNum(double val) {
        delegate = BigDecimal.valueOf(val);
    }
```

**Comment by nimo23** [comment]
How can I rewrite this

`estimateString = String.format("%1." + precision + "e", estimate);  (line 465)`

to something like:

`estimateString = String.format("%1.e", precision, estimate);  `

**Comment by nimo23** [comment]
Why is MathContext used for float?

```
private PrecisionNum(float val) {
        mathContext = new MathContext(DEFAULT_PRECISION, RoundingMode.HALF_UP);
        delegate = new BigDecimal(val, mathContext); // BigDecimal.valueOf(val); ?
    }

    private PrecisionNum(double val) {
        mathContext = new MathContext(DEFAULT_PRECISION, RoundingMode.HALF_UP);
        delegate = BigDecimal.valueOf(val);
    }
```

**Comment by milczarekIT** [comment]
> MathContext for float ok?

MathContext is for BigDecimal (output), float is an input

File: ta4j-core/src/main/java/org/ta4j/core/num/PrecisionNum.java [hunk 1]
```diff
@@ -744,7 +743,7 @@ public Num pow(Num n) {
         // use double pow(double, double)
         double xpowb = Math.pow(delegate.doubleValue(), bDouble);
         // use PrecisionNum.multiply(PrecisionNum)
-        BigDecimal result = xpowa.multiply(new BigDecimal(xpowb));
```
Comment (team172011) [inline ta4j-core/src/main/java/org/ta4j/core/num/PrecisionNum.java:747]: I am not sure if replacing `new` with `valueOf ` does make sense here. Did you compare the results or why are you changing this? Using new or valueOf do not the same and in this case I can imagine that cutting off digits (that is what the valueOf method does) is not the preffered behaviour.

File: ta4j-core/src/main/java/org/ta4j/core/num/PrecisionNum.java [hunk 2]
```diff
@@ -136,7 +136,7 @@ private PrecisionNum(BigDecimal val, int precision) {
      * @return the {@code Num}
      */
     public static PrecisionNum valueOf(String val) {
-        if (val.toUpperCase().equals("NAN")) {
```
Comment (team172011) [inline ta4j-core/src/main/java/org/ta4j/core/num/PrecisionNum.java:139]: Why do you want to change this? `toUpperCase `+ `equals` is not clearer than just one method call `equalsIgnoreCase`. Also the performance of `equalsIgnoreCase` is a little bit better.

**Comment by nimo23** [comment]
@team172011 yes, you are right. this PR does not make any sense. There is no question that PrecicionNum can be improved..but this PR does not. So I close it.

File: ta4j-core/src/main/java/org/ta4j/core/num/PrecisionNum.java (hunk 1 above)
Comment (nimo23) [reply ta4j-core/src/main/java/org/ta4j/core/num/PrecisionNum.java:747]: Yes, you are right. Did not know that..

File: ta4j-core/src/main/java/org/ta4j/core/num/PrecisionNum.java (hunk 2 above)
Comment (nimo23) [reply ta4j-core/src/main/java/org/ta4j/core/num/PrecisionNum.java:139]: ? `val.toUpperCase().equals` was the code **before**..I wanted to replace to `equalsIgnoreCase`..

**Pull request closed without merging**